corehq/apps/case_search/tests/test_filter_dsl.py

```diff
@@ -122,6 +124,186 @@ def test_simple_fuzzy_filter(self):
+
+        with patch("corehq.apps.case_search.filter_dsl._do_parent_lookup") as parent_lookup:
+            parent_lookup.return_value = ["123"]
+            built_filter = build_filter_from_ast(parsed, SearchFilterContext("domain", {"name"}))
+
+        self.checkQuery(built_filter, expected_filter, is_raw_query=True)
+        domain_arg, parent_built_filter, raw_query = parent_lookup.call_args.args
+        self.assertEqual(domain_arg, "domain")
+        self.assertEqual(raw_query, 'name = "farid"')
+
+        expected_parent_filter = {
+            "bool": {
+                "should": [
+                    {
+                        "nested": {
+                            "path": "case_properties",
+                            "query": {
+                                "bool": {
+                                    "filter": [
+                                        {
+                                            "term": {
+                                                "case_properties.key.exact": "name"
+                                            }
+                                        }
+                                    ],
+                                    "must": {
+                                        "match": {
+                                            "case_properties.value": {
+                                                "query": "farid",
+                                                "operator": "or",
+                                                "fuzziness": "AUTO"
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    },
+                    {
+                        "nested": {
+                            "path": "case_properties",
+                            "query": {
+                                "bool": {
+                                    "filter": [
+                                        {
+                                            "term": {
+                                                "case_properties.key.exact": "name"
+                                            }
+                                        }
+                                    ],
+                                    "must": {
+                                        "match": {
+                                            "case_properties.value": {
+                                                "query": "farid",
+                                                "operator": "or",
+                                                "fuzziness": "0"
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                ]
+            }
+        }
```

Comment:
random thought: I find these large JSON queries difficult to read, so in my most recent PR in the area, I used `es_query` functions to generate them instead.  So it's then comparing the output of an xpath expression with that of a `es_query` call.  Maybe an unneeded level of indirection, but something to consider if you feel likewise.

Reply:
I was wondering how to reduce the verbosity of these. I'll try that approach.

Reply:
18bfe27d66e